modules/patching-ovnk-address-ranges.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="patching-ovnk-address-ranges_{context}"]
+= Patching OVN-Kubernetes address ranges
+
+OVN-Kubernetes reserves the following IP address ranges:
+
+* `100.64.0.0/16`. This IP address range is used for the `internalJoinSubnet` parameter of OVN-Kubernetes by default. 
+* `100.88.0.0/16`. This IP address range is used for the `internalTransSwitchSubnet` parameter of OVN-Kubernetes by default.
+
+If OpenShift SDN has been configured to use either of these IP address ranges, you must patch them to use a different IP address range before initiating the limited live migration.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Enter the following command to retrieve the names of nodes in your cluster along with the IP address ranges that those nodes use:
++
+[source,terminal]
+----
+$ oc get nodes -o jsonpath='{range.items[*]}{.metadata.name}{"\t"}{.metadata.annotations.k8s\.ovn\.org/node-gateway-router-lrp-ifaddr}{"\n"}'
```

Comment:
This node annotation will only be added/used by OVN-K. You cannot find this in an SDN cluster. Users shall check all the networks that connect the cluster directly or indirectly. To make sure there is no IP range overlapping. For openshiftSDN, users only need to check if the `clusterNetwork` and `serviceNetwork` are overlapped with the OVN-K default internal subnets.

Reply:
Yeah. I agree.

Reply:
for `clusterNetwork` CIDR and `serviceNetwork` CIDR check,   there is pre-flight migration validation.  if there is overlaps with them.  The cluster will be blocked to migrate to OVN and CNO can expose the error.  So like no need to check 
`
$ oc get network/cluster -o jsonpath='{.status.clusterNetwork[*]}'

$ oc get networks.config/cluster -o jsonpath='{$.status.serviceNetwork}'
`

Reply:
@pliurh @zhaozhanqi When is the "pre-flight migration validation" initiated? That would change *when* users will patch 100.64.0.0/16 and 100.88.0.0/16. Is it when we initiate the limited live migration?

Reply:
@stevsmit 

> When is the "pre-flight migration validation" initiated? 

pre-flight migration validation will be run during migration.  that means once this command `oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'`,  it will check first if there is overlap CIDR. 

> That would change when users will patch 100.64.0.0/16 and 100.88.0.0/16. Is it when we initiate the limited live migration

for user want to patch CIDR, then firstly step patch, then do migration. The steps will be like:

# oc patch network.operator.openshift.io cluster --type='merge' -p='{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"ipv4":{"internalJoinSubnet": "100.64.2.0/16"}}}}}'
# oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'

Reply:
I think I have it now. PTAL.

Reply:
I also made Patching OVNK Address Ranges an optional procedure they can use if the migration is blocked during the pre-flight migration check: https://78178--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#patching-ovnk-address-ranges_migrate-from-openshift-sdn